core/indigo-core/molecule/meta_commons.h

```diff
@@ -138,54 +281,319 @@ namespace indigo
+
+        struct KETTextParagraph
         {
+            KETTextParagraph() : font_styles{}, alignment{}, indent{}, line_starts{}
+            {
+            }
             std::string text;
-            std::map<std::size_t, FONT_STYLE_SET> styles;
+            FONT_STYLE_SET font_style;
+            std::optional<TextAlignment> alignment;
+            std::optional<float> indent;
+            std::optional<std::set<int>> line_starts;
+            std::map<std::size_t, FONT_STYLE_SET> font_styles;
         };
 
         static const std::uint32_t CID = "Simple text object"_hash;
 
-        SimpleTextObject(const Vec3f& pos, const Vec2f& sz, const std::string& content);
+        static void convertToSimpleTextStyle(const FONT_STYLE_SET& fss, SimpleTextStyle& sts)
+        {
+            for (auto& fs : fss)
+            {
+                switch (fs.first.getFontStyle())
+                {
+                case KETFontStyle::FontStyle::EBold:
+                    sts.styles.push_back(KFontBoldStrV1);
+                    break;
+                case KETFontStyle::FontStyle::EItalic:
+                    sts.styles.push_back(KFontItalicStrV1);
+                    break;
+                case KETFontStyle::FontStyle::ESubScript:
+                    sts.styles.push_back(KFontSubscriptStrV1);
+                    break;
+                case KETFontStyle::FontStyle::ESuperScript:
+                    sts.styles.push_back(KFontSuperscriptStrV1);
+                    break;
+                case KETFontStyle::FontStyle::ESize: {
+                    auto fs_val = fs.first.getUInt();
+                    if (fs_val.has_value())
+                        sts.styles.push_back(std::string(KFontCustomSizeStrV1) + "_" + std::to_string(fs_val.value()) + "px");
+                    break;
+                }
+                default:
+                    break;
+                }
+            }
+        }
+
+        static void applyDispatcher(const rapidjson::Value& val, const DispatchMapKVP& disp_map)
+        {
+            for (auto kvp_it = val.MemberBegin(); kvp_it != val.MemberEnd(); ++kvp_it)
+            {
+                auto disp_it = disp_map.find(kvp_it->name.GetString());
+                if (disp_it != disp_map.end())
+                    disp_it->second(kvp_it->name.GetString(), kvp_it->value);
+            }
+        }
+
+        static void applyDispatcher(const rapidjson::Value& val, const DispatchMapVal& disp_map)
+        {
+            for (auto kvp_it = val.MemberBegin(); kvp_it != val.MemberEnd(); ++kvp_it)
+            {
+                auto disp_it = disp_map.find(kvp_it->name.GetString());
+                if (disp_it != disp_map.end())
+                    disp_it->second(kvp_it->value);
+            }
+        }
+
+        static auto floatLambda(std::optional<float>& val)
+        {
+            return [&val](const std::string&, const rapidjson::Value& float_val) {
+                if (float_val.IsFloat())
```

Comment:
Maybe we need same check for Int/String/Bool lambdas?

Reply:
fixed